src/commands/mod.rs

```diff
@@ -275,6 +315,43 @@ where
     map.end()
 }
 
+fn deserialize_option_string_not_empty<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt: Option<String> = Option::deserialize(deserializer)?;
+
+    match opt {
+        None => Ok(None),
+        Some(ref s) if s.is_empty() => Err(serde::de::Error::custom("String cannot be empty")),
+        _ => Ok(opt),
+    }
+}
+
+
+fn serialize_setting_type<S>(setting_type: &SettingType, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let s = match setting_type {
```

Comment:
just call to_string(0) (Implement display trait or auto derive it and then convert to lower case)